plugins/kaggle/api_token.go

```diff
@@ -0,0 +1,51 @@
+package kaggle
+
+import (
+	"github.com/1Password/shell-plugins/sdk"
+	"github.com/1Password/shell-plugins/sdk/importer"
+	"github.com/1Password/shell-plugins/sdk/provision"
+	"github.com/1Password/shell-plugins/sdk/schema"
+	"github.com/1Password/shell-plugins/sdk/schema/credname"
+	"github.com/1Password/shell-plugins/sdk/schema/fieldname"
+)
+
+func APIToken() schema.CredentialType {
+	return schema.CredentialType{
+		Name:          credname.APIToken,
+		DocsURL:       sdk.URL("https://www.kaggle.com/docs/api"),
+		ManagementURL: sdk.URL("https://www.kaggle.com/settings/account"),
+		Fields: []schema.CredentialField{
+			{
+				Name:                fieldname.Token,
+				MarkdownDescription: "API Token used to authenticate to Kaggle.",
+				Secret:              true,
+				Composition: &schema.ValueComposition{
+					Length: 32,
+					Charset: schema.Charset{
+						Lowercase: true,
+						Digits:    true,
+					},
+				},
+			},
+			{
+				Name:                fieldname.Username,
+				MarkdownDescription: "Username to authenticate to Kaggle.",
+				Secret:              true,
+				Composition: &schema.ValueComposition{
+					Charset: schema.Charset{
+						Lowercase: true,
+						Digits:    true,
+					},
+				},
+			},
+		},
+		DefaultProvisioner: provision.EnvVars(defaultEnvVarMapping),
+		Importer: importer.TryAll(
```

Comment:
It seems that Kaggle CLI looks for the configuration file at these paths `~/.kaggle/kaggle.json` on MacOS and Linux, and `C:\Users\<Windows-username>\.kaggle\kaggle.json` on Windows. The latter is not supported by shell plugins, so we can write one importer for both Mac and Linux.

Docs: https://www.kaggle.com/docs/api

Could you give it a try?